ERCS/erc-6774.md

```diff
@@ -0,0 +1,254 @@
+
+The value equivalence is so maintained by tokens creators as a whitelist, keeping these authorisations at the contract level is important and would help create a network of barterable NSTs.
+
+Based on value equivalence we leverage the game theory to enforce the non-sellability instead of stronger restrictions or centralization which would lead to higher code complexity and poor interoperability. Indeed, the strong counterpart at the transfer made NSTs highly illiquid to be subject to a sale on a marketplace or by over the counter (OTC) agreement.
+
+#### One way permissionless
+
+##### On `transferFor`
+
+Leaving this method permissionless would lead to the possibility to self transfer the token to any other address, without a barter.
+
+##### On `barter`
+
+Using `barter` without restriction opens up the risk of unwanted barters between non-equivalent NSTs. Thus any valueless NST could be used to perform a one-way transfer.
+
+### Multi NST contract barter
+
+Implementing a many-to-many NST contract is quite impossible as the barter occurs at the contract level, therefore implement such feature require to call and write on each NST contract. The gas usage for such a feature would be too expensive.
+
+### Register of allowed NST
+
+Creating a universal register for barter allowances between NST contracts (like the [ERC-1820](./erc-1820.md)) would be hard to maintain and easy to bypass.
+
+### Third party protocol
+
+Implementing a service for barter NFT is already existent (see Sudoswap OTC). Such a service still enable one-way transfer through ERC-721 properties and thus not prevent the sellability of the token.
+
+## Backwards Compatibility
+
+Excepted unimplemented methods specified in [Contract interface](#contract-interface), this EIP is backward compatible with [ERC-721](./erc-721.md).
+
+## Test Cases
+
+The reference implementation includes unit tests written using Foundry
+
+## Reference Implementation
+
+[Implementations examples](../assets/erc-6774/README.md) have been designed for this proposal.
+
+## Security Considerations
+
+### Sign and broadcast signed messages
+
+In regards to [EIP-712 security considerations](./eip-712.md#security-considerations), we include in the [Specification](#specification) that signed including barter terms must include a `nonce` to avoid replay attack, `tokenAddr` and the `tokenId` to avoid front-running attack on a barter.
+
+Moreover adding a `deadline` is also recommended in case of non-used signed message.
+
+### Verify the signature
+
+Using `ECDSA.sol`(`v4.7.3` at least) and `EIP-712.sol` libraries from OpenZeppelin are very recommended for signature verifications, verification only based on `ecrecover` should not be used (see the Solidity documentation, Mathematical and Cryptographic Functions).
```

Comment:
These are external links, even if they aren't hyperlinks. Please remove them, and instead include specific recommendations (like detecting malleable signatures.)